README.md

```diff
@@ -21,10 +24,23 @@ make html
 make serve
 ```
 
+5. Before committing your changes, install [`pre-commit`](https://pre-commit.com/),
+update the hooks, and run with
+
+```bash
+pre-commit install
+pre-commit autoupdate
+pre-commit run --all-files
+```
+
+to ensure that there are no style errors before committing your changes. The
+current hooks provide support for checking spelling errors and typos, Markdown
+style, and common file-based issues.
+
```

Comment:
`pre-commit` can automatically run before every commit locally ("pre commit" haha). I'd suggest adding something like this for the contributors -

https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit

Also, auto-update should not be used locally. This might update a particular pre-commit and that update might trigger several changes that might not be relevant to that particular PR.

Reply:
That makes sense. I am opening another PR for adding the changes you suggest